src/EShop.DAL/Interfaces/ICategoryRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using EShop.DAL.Entities;
+
+namespace EShop.DAL.Interfaces;
+
+public interface ICategoryRepository
+{
+    Task<Category> AddCategory(Category сategory);
+    Task<Category> UpdateCategory(Category сategory);
+    Task<Category> GetCategoryById(int id);
+    Task DeleteCategoryById(int id);
+    
+    Task<IEnumerable<Category>> GetAllCategory();
+}
```

Comment:
define IBaseRepository interface and inherit from it. define abstract class BaseRepository with basic methods